### PR TITLE
feat(node): add http and ws inbound transport

### PR DESCRIPTION
**Pull request by TimoGlastra**
Moves http and ws inbound transports to the node package. I know there's still work to be done on the transports, but this makes it 1000x easier to set up an agent. They're in the node package, so not strictly tied to core

### CODE REVIEW

**Comment by TimoGlastra** [comment]
@karimStekelenburg Once this is merged you can directly use the inbound transports

File: packages/node/src/transport/WsInboundTransport.ts [hunk 1]
```diff
@@ -1,22 +1,17 @@
...
+export class WsInboundTransport implements InboundTransporter {
+  private socketServer: Server
   private logger!: Logger
 
   // We're using a `socketId` just for the prevention of calling the connection handler twice.
   private socketIds: Record<string, unknown> = {}
 
-  public constructor(socketServer: WebSocket.Server) {
-    this.socketServer = socketServer
+  public constructor({ server, port }: { server: Server; port?: undefined } | { server?: undefined; port: number }) {
```
Comment (jakubkoci) [inline packages/node/src/transport/WsInboundTransport.ts:13]: Would it work with the following types?
```suggestion
  public constructor({ server, port }: { server: Server; port?: string } | { server?: Server; port: number }) {
```

**Review by jakubkoci — APPROVED** [review]
Yes, this is a good step forward 👍

File: packages/node/src/transport/WsInboundTransport.ts (hunk 1 above)
Comment (TimoGlastra) [reply packages/node/src/transport/WsInboundTransport.ts:13]: I want it to be an XOR statement. So either port or server should be provided but not both. If you create the server yourself, you set the port on the server. If you don't create the server yourself, you need to pass the port.

Could probably improve, but this was the easiest way I could think of with explicit typing that you shouldn't provide both

**Pull request merged**